cmd/pi-strings-go/utf16.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"encoding/binary"
+	"unicode"
+	"unicode/utf16"
+)
+
+// utf16DecodeRuneWithOrder unpacks the first UTF-16 encoding in p and returns
+// the rune and its width in bytes. The given byte order is used to decode
+// uint16 values of p. If p is empty it returns (ReplacementChar, 0). Otherwise,
+// if the encoding is invalid, it returns (ReplacementChar, 1)
+func utf16DecodeRuneWithOrder(p []byte, order binary.ByteOrder) (rune, int) {
+	var bs [2]uint16
+	n := 0
+	if len(p) >= 2 {
+		bs[0] = order.Uint16(p)
+		n++
+	}
+	if len(p) >= 4 {
+		bs[1] = order.Uint16(p[2:])
+		n++
+	}
+	return utf16DecodeRune(bs[:n])
+}
+
+// Copied from unicode/utf16 package of the Go 1.13 stdlib.
```

Comment:
I think we should take this f2f, since I don't have the UTF16 background to review this.

Reply:
Sure, lets do it tomorrow over a cup of tea :)